input/include/NovelRT/Input/NovelKey.hpp

```diff
@@ -27,9 +25,19 @@ namespace NovelRT::Input
 
         [[nodiscard]] inline bool operator==(NovelKey& other) noexcept
         {
-            return (_keyName == other.GetKeyName()) || (_pairedKey == other.GetExternalKeyCode());
+            return (_keyName == other.GetKeyName()) && (_pairedKey == other.GetExternalKeyCode());
         };
     };
+
+    inline bool operator==(NovelKey const& lhs, NovelKey const& rhs) noexcept
+    {
+        return lhs._pairedKey == rhs._pairedKey && (lhs._modifier == rhs._modifier) && lhs._keyName == rhs._keyName;
+    }
+
+    inline bool operator!=(NovelKey const& lhs, NovelKey const& rhs) noexcept
+    {
+        return lhs._pairedKey != rhs._pairedKey || (lhs._modifier == rhs._modifier) || lhs._keyName == rhs._keyName;
+    }
```

Comment:
We should avoid inlining functions as per our new header include policies. Are we updating the include structure for this in a separate submission? I don't remember.

Is there a reason we didn't use member operators for this? just curious.